nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -87,6 +91,13 @@ public static void checkDocumentEventContext(Event event, DocumentModel expected
 
         assertTrue(properties.containsKey(COMMENT));
         assertEquals(expectedCommentDocModel.getPropertyValue(COMMENT_TEXT), properties.get(COMMENT));
+
+        // The event source document must be the top level document as is the one linked in the notification.
+        assertTrue(properties.containsKey(TOP_LEVEL_DOCUMENT));
+        DocumentModel topLevelDocument = (DocumentModel) properties.get(TOP_LEVEL_DOCUMENT);
+        assertEquals(expectedTopLevelDocumentModel.getRef(), topLevelDocument.getRef());
```

Comment:
I'm wondering, we could maybe just check the equality between both documents, as the `DocumentModelImpl#equals` method checks the `id`s... WDYT @efge?

```suggestion
        assertEquals(expectedTopLevelDocumentModel, topLevelDocument);
```
Same for the next `assertEquals`.

Reply:
Can we garantee everybody uses this implementation ?

Reply:
In your test, it will always be `DocumentModelImpl` I hope :)

But keep it as it, do not change it 👌